src/utils/index.coffee

```diff
@@ -218,18 +229,66 @@ findKnownImportInDirectory = ({
     context
   }
 
-  foundExact = directoryCache.get name
-  foundCaseInsensitive = if (
-    settings['known-imports/case-insensitive-whitelist-filename']
+  foundExact = directoryCache.get(name) ? []
+  foundCaseInsensitive = (
+    if settings['known-imports/case-insensitive-whitelist-filename']
+      directoryCache.get(name.toLowerCase()) ? []
+    else
+      []
+  ).filter ({type}) -> type is 'filename'
+  hasAmbiguousMatches =
+    foundExact.length > 1 or
+    (not foundExact.length and foundCaseInsensitive.length > 1)
```

Comment:
I'm not considering a combination of "exact" + "case-insensitive" matches to be an "ambiguous" multiple-matches situation (exact wins, so if there's just a single exact match + some case-insensitive matches then the exact match will always be used)